### PR TITLE
basecamp: init at 2.3.8

### PR DESCRIPTION
**Pull request by yamashitax**
###### Description of changes
Add package [Basecamp](https://basecamp.com/)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/basecamp/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{ lib
+, fetchurl
+, undmg
+, stdenv
+, metaCommon ? { }
```
Comment (ocfox) [inline pkgs/applications/misc/basecamp/default.nix:5]:
```suggestion
```

File: pkgs/applications/misc/basecamp/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  installPhase = ''
+    mkdir -p "$out/Applications/Basecamp 3.app"
+    cp -R . "$out/Applications/Basecamp 3.app"
+    mkdir "$out/bin"
+    ln -s "$out/Applications/Basecamp 3.app/Contents/MacOS/Basecamp 3" "$out/bin/basecamp"
+  '';
+
+  meta = with lib; {
+    description = "Refreshingly simple project management.";
```
Comment (ocfox) [inline pkgs/applications/misc/basecamp/default.nix:38]:
```suggestion
    description = "Refreshingly simple project management";
```

File: pkgs/applications/misc/basecamp/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    homepage = "https://basecamp.com";
+    license = licenses.free;
```
Comment (ocfox) [inline pkgs/applications/misc/basecamp/default.nix:40]:
```suggestion
    license = licenses.unfree;
```

**Review by ocfox — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/basecamp/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, fetchurl
+, undmg
+, stdenv
+}:
+
+stdenv.mkDerivation rec {
```
Comment (drupol) [inline pkgs/applications/misc/basecamp/default.nix:7]: I think we should use the `finalAttrs` pattern here.

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

**Review by drupol — CHANGES_REQUESTED** [review]

**Comment by yamashitax** [comment]
Closing this, since we can not get reproducible builds. There is nothing denoting any specific versions in the download urls, so as soon as a new version is available the build will fail.

**Pull request closed without merging**